jenkins-plugin/src/test/java/jenkins/scm/impl/mock/GitRepoRule.java

```diff
@@ -0,0 +1,120 @@
+package jenkins.scm.impl.mock;
+
+import com.gargoylesoftware.htmlunit.WebResponse;
+import com.gargoylesoftware.htmlunit.util.NameValuePair;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.eclipse.jgit.lib.Constants;
+import org.eclipse.jgit.lib.RepositoryBuilder;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.FileVisitOption;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.EnumSet;
+
+import static java.nio.file.FileVisitResult.*;
+
+/**
+ * @author <a href="mailto:cleclerc@cloudbees.com">Cyrille Le Clerc</a>
+ */
+public class GitRepoRule extends AbstractSampleDVCSRepoRule {
```

Comment:
`GitSampleRepoRule` is final and its `init`method creates a file `file` at the root level that I prefer to not have in my git repo.

For these reasons, I temporarily created my own, this is a WIP waiting to better understand what is best.

Moreover, my desire to make this GitRepoRule work on remote agent  let me think that I may look at adapting `jenkins.scm.impl.mock.AbstractSampleRepoRule#run` to make it execute commands remotely.